app/src/main/java/com/github/swent/echo/compose/components/HomeScreen.kt

```diff
@@ -8,118 +8,105 @@ import androidx.compose.material3.ModalNavigationDrawer
 import androidx.compose.material3.Scaffold
 import androidx.compose.material3.rememberDrawerState
 import androidx.compose.runtime.Composable
-import androidx.compose.runtime.MutableState
-import androidx.compose.runtime.mutableStateOf
-import androidx.compose.runtime.remember
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
 import androidx.compose.runtime.rememberCoroutineScope
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.platform.testTag
-import com.github.swent.echo.compose.components.searchmenu.FiltersContainer
-import com.github.swent.echo.compose.components.searchmenu.SortBy
 import com.github.swent.echo.compose.map.MapDrawer
-import com.github.swent.echo.data.SAMPLE_EVENTS
-import com.github.swent.echo.data.model.Event
 import com.github.swent.echo.ui.navigation.NavigationActions
-import java.time.ZonedDateTime
+import com.github.swent.echo.viewmodels.HomeScreenViewModel
+import com.github.swent.echo.viewmodels.MapOrListMode
+import com.github.swent.echo.viewmodels.Overlay
 
-enum class Overlay {
-    NONE,
-    EVENT_INFO_SHEET,
-    SEARCH_SHEET
-}
-
-enum class MapOrListMode {
-    MAP,
-    LIST
-}
 /**
  * The scaffold for the home screen. contains the top bar, the bottom sheet if an event is selected,
  * the hamburger menu when the button is clicked on the top bar, the search bottom sheet, the list
  * view, and the map.
  */
 @Composable
-fun HomeScreen(navActions: NavigationActions) {
-    // State for the MapOrListMode
-    val overlay = remember { mutableStateOf(Overlay.NONE) }
-    val mode = remember { mutableStateOf(MapOrListMode.MAP) }
-    val displayEventInfo = remember { mutableStateOf<Event?>(null) }
+fun HomeScreen(navActions: NavigationActions, homeScreenViewModel: HomeScreenViewModel) {
+
     // Drawer state to open and close the hamburger menu
     val drawerState = rememberDrawerState(initialValue = DrawerValue.Closed)
     val scope = rememberCoroutineScope()
+    val mode by homeScreenViewModel.mode.collectAsState()
+
     // Hamburger menu compose
     ModalNavigationDrawer(
         // Content of the hamburger menu
-        drawerContent = { HamburgerMenuDrawerSheet(navActions, drawerState, scope, {}) },
+        drawerContent = {
+            HamburgerMenuDrawerSheet(navActions, drawerState, scope) {
+                homeScreenViewModel.signOut()
```

Comment:
Here, we should call `navActions.navigateTo(Routes.LOGIN)` to redirect the user to the login screen.

Reply:
makes sense